os/board/esp32_DevKitC/src/esp32_boot.c

```diff
@@ -0,0 +1,364 @@
+	if (ret < 0) {
+		lldbg("Failed to mount procfs at %s: %d\n", CONFIG_ESP32_PROCFS_MOUNTPOINT, ret);
+	}
+#endif
+
+}
+#endif
+
+void esp32_board_initialize(void)
+{
+}
+
+static void board_gpio_initialize(void)
+{
+#ifdef CONFIG_GPIO
+	int i;
+	struct gpio_lowerhalf_s *lower;
+
+	struct {
+		uint8_t minor;
+		uint16_t pincfg;
+	} pins[] = {
+		{
+			5, INPUT_PULLDOWN
+		}, {
+			15, INPUT_PULLDOWN
+		}, {
+			18, INPUT_PULLDOWN
+		}, {
+			19, INPUT_PULLDOWN
+		}, {
+			21, INPUT_PULLDOWN
+		},
+	};
+
+	for (i = 0; i < sizeof(pins) / sizeof(*pins); i++) {
+		lower = esp32_gpio_lowerhalf(pins[i].minor, pins[i].pincfg);
+		gpio_register(pins[i].minor, lower);
+	}
+#endif
+}
+
+static void board_i2c_initialize(void)
+{
+#if defined(CONFIG_I2C)
+#if defined(CONFIG_ESP32_I2C0) && (1 == CONFIG_ESP32_I2C0)
+	esp32_i2c_register(0);
+#endif
+
+#if defined(CONFIG_ESP32_I2C1) && (1 == CONFIG_ESP32_I2C1)
+	esp32_i2c_register(1);
+#endif
+#endif
+}
+
+#if defined(CONFIG_ADC)
+static void board_adc_initialize(void)
+{
+    static bool adc_initialized;
+    struct adc_dev_s* adc;
+    const adc_channel_t chanlist[] = {
+        ADC_CHANNEL_4,
+        ADC_CHANNEL_5,
+        ADC_CHANNEL_6,
+        ADC_CHANNEL_7
+    };
+
+    /* Have we already initialized? */
+    if (!adc_initialized) {
+
+        adc = esp32_adc_initialize(chanlist, sizeof(chanlist) / sizeof(adc_channel_t));
+        if (NULL != adc) {
+            lldbg("ERROR: Failed to get the ESP32 ADC driver\n");
+        }
+        else {
+            lldbg("ERROR: up_spiinitialize failed\n");
+        }
+
+        /* Register the ADC driver at "/dev/adc0" */
+        int ret = adc_register("/dev/adc0", adc);
+        if (ret < 0) {
+            return;
+        }
+
+        /* Now we are initialized */
+
+        adc_initialized = true;
+    }
+}
+#endif
+
+/****************************************************************************
+ * Name: board_initialize
+ *
+ * Description:
+ *   If CONFIG_BOARD_INITIALIZE is selected, then an additional
+ *   initialization call will be performed in the boot-up sequence to a
+ *   function called board_initialize().  board_initialize() will be
+ *   called immediately after up_initialize() is called and just before the
+ *   initial application is started.  This additional initialization phase
+ *   may be used, for example, to initialize board-specific device drivers.
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_BOARD_INITIALIZE
+void board_initialize(void)
+{
+	/* Perform board-specific initialization */
+	(void)esp32_bringup();
+#if defined(CONFIG_ESP32_FLASH_PART) && (CONFIG_ESP32_FLASH_PART == 1)	
+	esp32_devKit_config_partions();
+	esp32_devKit_mount_partions();
+#endif
+	board_gpio_initialize();
+	board_i2c_initialize();
+	board_ledc_setup();
+
+#if defined(CONFIG_ADC)
+	board_adc_initialize();
+#endif
+
+/*Init SPI RAM*/
+#ifdef CONFIG_SPIRAM_SUPPORT
+    esp_spiram_init_cache();
```

Comment:
Please check violations of coding rules.
Indentation (start of line) should be a tab, not spaces.

Reply:
Fixed, thanks.